src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -71,12 +76,12 @@ USBD_CDC_LineCodingTypeDef LineCoding =
   0x08    /* nb. of bits 8*/
 };
 
-uint8_t UserRxBuffer[APP_RX_DATA_SIZE];/* Received Data over USB are stored in this buffer */
-uint8_t UserTxBuffer[APP_TX_DATA_SIZE];/* Received Data over UART (CDC interface) are stored in this buffer */
+volatile uint8_t UserRxBuffer[APP_RX_DATA_SIZE];/* Received Data over USB are stored in this buffer */
+volatile uint8_t UserTxBuffer[APP_TX_DATA_SIZE];/* Received Data over UART (CDC interface) are stored in this buffer */
 uint32_t BuffLength;
-uint32_t UserTxBufPtrIn = 0;/* Increment this pointer or roll it back to
+volatile uint32_t UserTxBufPtrIn = 0;/* Increment this pointer or roll it back to
```

Comment:
`ATOMIC_BLOCK` is memory barrier, all variables used init are reread from memory (but then cached in register) and written on block exit. So using `ATOMIC_BLOCK` in `HAL_TIM_PeriodElapsedCallback` will make `volatile` unnecessary (and as long as HAL_TIM_PeriodElapsedCallback is called in ISR context, it will work even without ATOMIC_BLOCK, but using it will make access to shared data explicit)